core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1033,10 +1086,22 @@ public long createFile(AlluxioURI path, CreateFileOptions options)
       BlockInfoException, IOException, FileDoesNotExistException {
     Metrics.CREATE_FILES_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
-        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
-      mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE);
+        MasterAuditContext auditContext = createAuditContext("createFile", path, null, null)) {
+      if (options.isRecursive()) {
```

Comment:
HDFS obtains a list of inodes along the path and logs the owner:group:mode of last existing inode in the path.
In Alluxio, if the creation is recursive, we initially record the permission of the last existing inode in the path. If the creation fails, we have this piece of information to assist debugging.
If the creation is NOT recursive, we initially check the information of parent. If the parent does not exist, then the creation should fail and throw FileDoesNotExistException. The audit log entry for this event should be null.
I'm thinking, if the operation succeeds, should we still log the permission of parent?